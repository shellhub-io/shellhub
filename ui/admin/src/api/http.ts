import { AxiosInstance } from "axios";
import { setupInterceptorsTo } from "./interceptors";
import * as axiosTs from "./client";
import { Configuration } from "./client";
import { BaseAPI } from "./client/base";

const configuration = new Configuration();
configuration.basePath = `${window.location.protocol}//${window.location.host}`;
configuration.accessToken = localStorage.getItem("cloud_token") || "";

// eslint-disable-next-line import/no-mutable-exports
let adminApi = new axiosTs.AdminApi(configuration);
// eslint-disable-next-line import/no-mutable-exports
let cloudApi = new axiosTs.CloudApi(configuration);

export const createNewClient = () => {
  const newConfiguration = new Configuration();
  newConfiguration.basePath = `${window.location.protocol}//${window.location.host}`;
  newConfiguration.accessToken = localStorage.getItem("cloud_token") || "";
  adminApi = new axiosTs.AdminApi(newConfiguration);
  cloudApi = new axiosTs.CloudApi(newConfiguration);
  return { adminApi, cloudApi };
};

declare module "./client/base" {
  interface BaseAPI {
    getAxios(): AxiosInstance;
    getConfiguration(): Configuration | undefined;
    setConfiguration(configuration: Configuration): void;
  }
}

/** Returns the axios instance */
BaseAPI.prototype.getAxios = function getAxios(this: BaseAPI): AxiosInstance {
  return this.axios;
};

/** Returns the configuration */
// eslint-disable-next-line vue/max-len
BaseAPI.prototype.getConfiguration = function getConfiguration(this: BaseAPI): Configuration | undefined {
  return this.configuration;
};

/** Sets the configuration */
// eslint-disable-next-line vue/max-len
BaseAPI.prototype.setConfiguration = function setConfiguration(this: BaseAPI, configuration: Configuration): void {
  this.configuration = configuration;
};

// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
setupInterceptorsTo(adminApi.axios);

export { adminApi, cloudApi };
