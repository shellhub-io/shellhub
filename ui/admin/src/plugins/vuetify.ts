// Styles
// import "@mdi/font/css/materialdesignicons.css";
import "vuetify/styles";

// Vuetify
import { createVuetify } from "vuetify";

export default createVuetify({
  theme: {
    defaultTheme: "dark",
  },
});
