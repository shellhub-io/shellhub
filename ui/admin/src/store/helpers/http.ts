import Axios from "axios";
// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore //* use while route mapping is not in TypeScript
export default () => {
  const axios = Axios.create({
    baseURL: `${window.location.protocol}//${window.location.host}/admin`,
    headers: {
      Authorization: `Bearer ${localStorage.getItem("cloud_token")}`,
    },
  });

  return axios;
};
