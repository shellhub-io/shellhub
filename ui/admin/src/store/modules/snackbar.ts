import { Module } from "vuex";
import { State } from "./../index";

export interface snackbarMessageAndContentType {
  typeMessage: string;
  typeContent: string;
}

export interface SnackbarState {
  snackbarError: boolean;
  snackbarSuccess: boolean;
  snackbarMessageAndContentType: snackbarMessageAndContentType;
  snackbarCopy: boolean;
}

export const snackbar: Module<SnackbarState, State> = {
  namespaced: true,

  state: {
    snackbarError: false,
    snackbarSuccess: false,
    snackbarMessageAndContentType: { typeMessage: "", typeContent: "" },
    snackbarCopy: false,
  },

  getters: {
    snackbarSuccess: (state) => state.snackbarSuccess,
    snackbarError: (state) => state.snackbarError,
    snackbarMessageAndContentType: (state) => state.snackbarMessageAndContentType,
    snackbarCopy: (state) => state.snackbarCopy,
  },

  mutations: {
    setSnackbarSuccessAction: (state, data) => {
      state.snackbarMessageAndContentType = data;
      state.snackbarSuccess = true;
    },

    setSnackbarNoContent: (state, data) => {
      state.snackbarMessageAndContentType = data;
      state.snackbarSuccess = true;
    },

    setSnackbarSuccessDefault: (state) => {
      state.snackbarMessageAndContentType = { typeMessage: "default", typeContent: "" };
      state.snackbarSuccess = true;
    },

    unsetSnackbarSuccess: (state) => {
      state.snackbarSuccess = false;
    },

    setSnackbarErrorLoadingOrAction: (state, data) => {
      state.snackbarMessageAndContentType = data;
      state.snackbarError = true;
    },

    setSnackbarErrorCustom: (state, data) => {
      state.snackbarMessageAndContentType = data;
      state.snackbarError = true;
    },

    setSnackbarErrorDefault: (state) => {
      state.snackbarMessageAndContentType = { typeMessage: "default", typeContent: "" };
      state.snackbarError = true;
    },

    setSnackbarErrorLicense: (state, data) => {
      state.snackbarMessageAndContentType = data;
      state.snackbarError = true;
    },

    unsetSnackbarError: (state) => {
      state.snackbarError = false;
    },

    setSnackbarCopy: (state, value) => {
      state.snackbarMessageAndContentType = { typeMessage: "", typeContent: value };
      state.snackbarCopy = true;
    },

    unsetSnackbarCopy: (state) => {
      state.snackbarCopy = false;
    },
  },

  actions: {
    showSnackbarSuccessAction: ({ commit }, value) => {
      const data = { typeMessage: "action", typeContent: value };
      commit("setSnackbarSuccessAction", data);
    },

    showSnackbarSuccessDefault: ({ commit }) => {
      commit("setSnackbarSuccessDefault");
    },

    unsetShowStatusSnackbarSuccess: ({ commit }) => {
      commit("unsetSnackbarSuccess");
    },

    showSnackbarErrorLoading: ({ commit }, value) => {
      const data = { typeMessage: "loading", typeContent: value };
      commit("setSnackbarErrorLoadingOrAction", data);
    },

    showSnackbarNoContent: ({ commit }) => {
      const data = { typeMessage: "no-content", typeContent: "" };
      commit("setSnackbarNoContent", data);
    },

    showSnackbarErrorAction: ({ commit }, value) => {
      const data = { typeMessage: "action", typeContent: value };
      commit("setSnackbarErrorLoadingOrAction", data);
    },

    showSnackbarErrorCustom: ({ commit }, value) => {
      const data = { typeMessage: "custom", typeContent: value };
      commit("setSnackbarErrorCustom", data);
    },

    showSnackbarErrorLicense: ({ commit }, value) => {
      const data = { typeMessage: "licenseRequired", typeContent: value };
      commit("setSnackbarErrorLicense", data);
    },

    showSnackbarErrorDefault: ({ commit }) => {
      commit("setSnackbarErrorDefault");
    },

    unsetShowStatusSnackbarError: ({ commit }) => {
      commit("unsetSnackbarError");
    },

    showSnackbarCopy: ({ commit }, value) => {
      commit("setSnackbarCopy", value);
    },

    unsetShowStatusSnackbarCopy: ({ commit }) => {
      commit("unsetSnackbarCopy");
    },
  },
};
