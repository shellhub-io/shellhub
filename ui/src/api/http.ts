import axios, { AxiosInstance } from "axios";
import * as axiosTs from "./client";
import { Configuration } from "./client";
import { BaseAPI } from "./client/base";
import { setupInterceptorsTo } from "./interceptors";

// This is the default configuration for local instance endpoints
const configuration = new Configuration();
configuration.basePath = `${window.location.protocol}//${window.location.host}`;
configuration.accessToken = localStorage.getItem("token") || "";

// We need a custom configuration for cloud endpoints
const cloudApiConfiguration = new Configuration();
cloudApiConfiguration.basePath = "https://cloud.shellhub.io";

// Creates a new axios instance and setup interceptors by default
const newAxiosInstance = (setupInterceptor = true): AxiosInstance => {
  const instance = axios.create();
  if (setupInterceptor) setupInterceptorsTo(instance);
  return instance;
};
const sessionsApi = new axiosTs.SessionsApi(
  configuration,
  undefined,
  newAxiosInstance(),
);
const devicesApi = new axiosTs.DevicesApi(
  configuration,
  undefined,
  newAxiosInstance(),
);
const containersApi = new axiosTs.ContainersApi(
  configuration,
  undefined,
  newAxiosInstance(),
);
const systemApi = new axiosTs.SystemApi(
  configuration,
  undefined,
  newAxiosInstance(),
);
const namespacesApi = new axiosTs.NamespacesApi(
  configuration,
  undefined,
  newAxiosInstance(),
);
const apiKeysApi = new axiosTs.ApiKeysApi(
  configuration,
  undefined,
  newAxiosInstance(),
);
const sshApi = new axiosTs.SshApi(
  configuration,
  undefined,
  newAxiosInstance(),
);
const tagsApi = new axiosTs.TagsApi(
  configuration,
  undefined,
  newAxiosInstance(),
);
const usersApi = new axiosTs.UsersApi(
  configuration,
  undefined,
  newAxiosInstance(),
);
const mfaApi = new axiosTs.MfaApi(
  configuration,
  undefined,
  newAxiosInstance(),
);
const tunnelApi = new axiosTs.TunnelsApi(
  configuration,
  undefined,
  newAxiosInstance(),
);
const billingApi = new axiosTs.BillingApi(
  configuration,
  undefined,
  newAxiosInstance(),
);
const rulesApi = new axiosTs.RulesApi(
  configuration,
  undefined,
  newAxiosInstance(),
);
const announcementApi = new axiosTs.AnnouncementsApi(
  cloudApiConfiguration,
  undefined,
  newAxiosInstance(false),
);

/**
 * Extends the interface BaseAPI to include a new method `getAxios` allowing
 * access to the protected axios property without the need to access it
 * directly from outside the class and avoiding a linting error caused by
 * accessing a protected property.
 **/
declare module "./client/base" {
  interface BaseAPI {
    getAxios(): AxiosInstance;
    getConfiguration(): Configuration | undefined;
    setConfiguration(configuration: Configuration): void;
  }
}

/** Returns the axios instance */
BaseAPI.prototype.getAxios = function getAxios(this: BaseAPI): AxiosInstance {
  return this.axios;
};

/** Returns the configuration */
BaseAPI.prototype.getConfiguration = function getConfiguration(this: BaseAPI): Configuration | undefined {
  return this.configuration;
};

/** Sets the configuration */
BaseAPI.prototype.setConfiguration = function setConfiguration(this: BaseAPI, configuration: Configuration): void {
  this.configuration = configuration;
};

// Reloads the configuration for all APIs
const reloadConfiguration = () => {
  [
    sessionsApi,
    devicesApi,
    containersApi,
    systemApi,
    namespacesApi,
    apiKeysApi,
    sshApi,
    tagsApi,
    usersApi,
    mfaApi,
    tunnelApi,
    billingApi,
    rulesApi,
  ].forEach((api) => {
    api.setConfiguration(configuration);
  });
};

export {
  configuration,
  reloadConfiguration,
  sessionsApi,
  devicesApi,
  containersApi,
  systemApi,
  namespacesApi,
  apiKeysApi,
  sshApi,
  tagsApi,
  usersApi,
  mfaApi,
  tunnelApi,
  billingApi,
  rulesApi,
  announcementApi,
};
