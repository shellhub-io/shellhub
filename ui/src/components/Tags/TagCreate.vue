<template>
  <BaseDialog v-model="showDialog" min-width="300" max-width="600">
    <v-card class="bg-v-theme-surface">
      <v-card-title class="text-h5 pa-4 bg-primary"> Create Tag </v-card-title>
      <v-divider />

      <v-card-text class="mt-4 mb-0 pb-1">
        <v-text-field
          v-model="inputTags"
          label="Tag name"
          :error-messages="tagsError"
          required
          variant="underlined"
          data-test="tag-field"
        />
      </v-card-text>

      <v-card-actions>
        <v-spacer />
        <v-btn variant="text" data-test="close-btn" @click="close()">
          Close
        </v-btn>

        <v-btn
          color="primary"
          variant="text"
          data-test="create-btn"
          @click="create()"
        >
          Create
        </v-btn>
      </v-card-actions>
    </v-card>
  </BaseDialog>
</template>

<script setup lang="ts">
import { ref, computed, watch } from "vue";
import handleError from "@/utils/handleError";
import useSnackbar from "@/helpers/snackbar";
import useTagsStore from "@/store/modules/tags";
import BaseDialog from "../BaseDialog.vue";

const emit = defineEmits(["update"]);
const tagsStore = useTagsStore();
const snackbar = useSnackbar();
const showDialog = defineModel({ default: false });

const inputTags = ref<string>("");
const tagsError = ref("");
const tagsHasLessThan3Characters = computed(() => inputTags.value.length < 3);
const tenant = computed(() => localStorage.getItem("tenant"));

watch(inputTags, () => {
  if (inputTags.value.length > 255) {
    tagsError.value = "Maximum of 3 tags";
  } else if (tagsHasLessThan3Characters.value) {
    tagsError.value = "The minimum length is 3 characters";
  } else {
    tagsError.value = "";
  }
});

const close = () => {
  inputTags.value = "";
  showDialog.value = false;
};

const update = () => {
  emit("update");
  close();
};

const create = async () => {
  if (!tagsError.value) {
    try {
      await tagsStore.createTag({
        tenant: tenant.value || "",
        name: inputTags.value,
      });

      update();
      snackbar.showSuccess("Successfully created tag");
    } catch (error: unknown) {
      snackbar.showError("Failed to create tag.");
      handleError(error);
    }
  }
};

defineExpose({ inputTags, showDialog });
</script>
