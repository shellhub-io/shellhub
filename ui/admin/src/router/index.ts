import { createRouter, createWebHistory, NavigationGuardNext, RouteLocationNormalized } from "vue-router";
import Login from "@admin/views/Login.vue";
import Dashboard from "@admin/views/Dashboard.vue";
import Users from "@admin/views/Users.vue";
import SettingsLicense from "@admin/components/Settings/SettingsLicense.vue";
import SettingsAuthentication from "@admin/components/Settings/SettingsAuthentication.vue";
import Namespaces from "@admin/views/Namespaces.vue";
import Settings from "@admin/views/Settings.vue";
import { computed } from "vue";
import useLicenseStore from "@admin/store/modules/license";
import useLayoutStore from "@admin/store/modules/layout";
import useAuthStore from "@admin/store/modules/auth";
import { plugin as snackbar } from "@/plugins/snackbar"; // using direct plugin because inject() doesn't work outside components

const routes = [
  {
    path: "/login",
    name: "login",
    component: Login,
    meta: {
      layout: "SimpleLayout",
      requiresAuth: false,
    },
  },
  {
    path: "/",
    name: "dashboard",
    component: Dashboard,
  },
  {
    path: "/users",
    name: "users",
    component: Users,
  },
  {
    path: "/user/:id",
    name: "userDetails",
    component: () => import("@admin/views/UserDetails.vue"),
  },
  {
    path: "/devices",
    name: "devices",
    component: () => import("@admin/views/Device.vue"),
    redirect: {
      name: "listDevices",
    },
    children: [
      {
        path: "",
        name: "listDevices",
        component: () => import("@admin/views/Device.vue"),
      },
    ],
  },
  {
    path: "/device/:id",
    name: "deviceDetails",
    component: () => import("@admin/views/DeviceDetails.vue"),
  },
  {
    path: "/sessions",
    name: "sessions",
    component: () => import("@admin/views/Sessions.vue"),
  },
  {
    path: "/session/:id",
    name: "sessionDetails",
    component: () => import("@admin/views/SessionDetails.vue"),
  },
  {
    path: "/settings",
    name: "Settings",
    component: Settings,
    redirect: { name: "SettingProfile" },
    children: [
      {
        path: "authentication",
        name: "SettingAuthentication",
        component: SettingsAuthentication,
      },
      {
        path: "license",
        name: "SettingLicense",
        component: SettingsLicense,
      },
    ],
  },
  {
    path: "/firewall-rules",
    name: "firewall-rules",
    component: () => import("@admin/views/FirewallRules.vue"),
  },
  {
    path: "/firewall-rules/:id",
    name: "firewallRulesDetails",
    component: () => import("@admin/views/FirewallRulesDetails.vue"),
  },
  {
    path: "/namespaces",
    name: "namespaces",
    component: Namespaces,
  },
  {
    path: "/namespace/:id",
    name: "namespaceDetails",
    component: () => import("@admin/views/NamespaceDetails.vue"),
  },
  {
    path: "/announcements",
    name: "announcements",
    component: () => import("@admin/views/Announcements.vue"),
  },
  {
    path: "/announcement/:uuid",
    name: "announcementDetails",
    component: () => import("@admin/views/AnnouncementDetails.vue"),
  },
  {
    path: "/new-announcement",
    name: "new-announcement",
    component: () => import("@admin/views/NewAnnouncement.vue"),
  },
];

const router = createRouter({
  history: createWebHistory("/admin/"),
  routes,
});

router.beforeEach(
  async (to: RouteLocationNormalized, from: RouteLocationNormalized, next: NavigationGuardNext) => {
    const licenseStore = useLicenseStore();
    const layoutStore = useLayoutStore();
    const authStore = useAuthStore();

    const requiresAuth = to.meta.requiresAuth ?? true;

    const layout = to.meta.layout || "AppLayout";
    layoutStore.setLayout(layout as string);

    if (!authStore.isLoggedIn && requiresAuth) {
      return next({
        name: "login",
        query: { redirect: to.fullPath },
      });
    }

    if (authStore.isLoggedIn && !to.meta.requiresAuth) {
      const license = computed(() => licenseStore.getLicense);

      try {
        await licenseStore.get();

        if (license.value.expired && to.name !== "SettingLicense") {
          snackbar.showError("Your license has expired. Please update it and try again.");
          return next({ name: "SettingLicense" });
        }
      } catch {
        if (to.name !== "SettingLicense") {
          snackbar.showError("Failed to get your license info. Please check it and try again.");
          return next({ name: "SettingLicense" });
        }
      }
    }

    return next();
  },
);

export default router;
