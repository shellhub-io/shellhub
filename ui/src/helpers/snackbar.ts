import { inject } from "vue";
import { InjectionKey, type ISnackbarPlugin } from "@/plugins/snackbar";

const useSnackbar = (): ISnackbarPlugin => inject(InjectionKey) as ISnackbarPlugin;

export default useSnackbar;
