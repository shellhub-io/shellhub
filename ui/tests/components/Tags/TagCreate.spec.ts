import { flushPromises, DOMWrapper, mount, VueWrapper } from "@vue/test-utils";
import { createVuetify } from "vuetify";
import MockAdapter from "axios-mock-adapter";
import { expect, describe, it, beforeEach, vi } from "vitest";
import { createPinia, setActivePinia } from "pinia";
import TagCreate from "@/components/Tags/TagCreate.vue";
import { router } from "@/router";
import { tagsApi } from "@/api/http";
import { SnackbarInjectionKey } from "@/plugins/snackbar";
import useTagsStore from "@/store/modules/tags";

const node = document.createElement("div");
node.setAttribute("id", "app");
document.body.appendChild(node);

const devices = [
  {
    uid: "a582b47a42d",
    name: "39-5e-2a",
    identity: {
      mac: "00:00:00:00:00:00",
    },
    info: {
      id: "linuxmint",
      pretty_name: "Linux Mint 19.3",
      version: "",
    },
    public_key: "----- PUBLIC KEY -----",
    tenant_id: "fake-tenant-data",
    last_seen: "2020-05-20T18:58:53.276Z",
    online: false,
    namespace: "user",
    status: "accepted",
    tags: ["test"],
  },
  {
    uid: "a582b47a42e",
    name: "39-5e-2b",
    identity: {
      mac: "00:00:00:00:00:00",
    },
    info: {
      id: "linuxmint",
      pretty_name: "Linux Mint 19.3",
      version: "",
    },
    public_key: "----- PUBLIC KEY -----",
    tenant_id: "fake-tenant-data",
    last_seen: "2020-05-20T19:58:53.276Z",
    online: true,
    namespace: "user",
    status: "accepted",
    tags: ["test2"],
  },
];

const mockSnackbar = {
  showSuccess: vi.fn(),
  showError: vi.fn(),
};

describe("Tag Form Create", async () => {
  let wrapper: VueWrapper<InstanceType<typeof TagCreate>>;
  setActivePinia(createPinia());

  const tagsStore = useTagsStore();
  const vuetify = createVuetify();

  let mockTags: MockAdapter;

  beforeEach(async () => {
    const el = document.createElement("div");
    document.body.appendChild(el);

    localStorage.setItem("tenant", "fake-tenant-data");

    mockTags = new MockAdapter(tagsApi.getAxios());

    mockTags.onGet("http://localhost:3000/api/devices?filter=&page=1&per_page=10&status=accepted").reply(200, devices);

    wrapper = mount(TagCreate, {
      global: {
        plugins: [vuetify, router],
        provide: { [SnackbarInjectionKey]: mockSnackbar },
        config: {
          errorHandler: () => { /* ignore global error handler */ },
        },
      },
    });
  });

  it("Is a Vue instance", () => {
    expect(wrapper.vm).toBeTruthy();
  });

  it("Renders the component", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("Data is defined", () => {
    expect(wrapper.vm.$data).toBeDefined();
  });

  it("Renders the component table", async () => {
    const dialog = new DOMWrapper(document.body);
    wrapper.vm.showDialog = true;
    await flushPromises();
    expect(wrapper.findComponent('[data-test="tag-field"]').exists()).toBe(true);
    expect(dialog.find('[data-test="close-btn"]').exists()).toBe(true);
    expect(dialog.find('[data-test="create-btn"]').exists()).toBe(true);
  });

  it("Successfully create tag", async () => {
    wrapper.vm.showDialog = true;
    await flushPromises();

    mockTags.onPost("http://localhost:3000/api/namespaces/fake-tenant-data/tags").reply(200);

    const tagsSpy = vi.spyOn(tagsStore, "createTag");

    await wrapper.findComponent('[data-test="tag-field"]').setValue("tag-test2");

    await wrapper.findComponent('[data-test="create-btn"]').trigger("click");

    await flushPromises();

    expect(tagsSpy).toHaveBeenCalledWith({
      tenant: "fake-tenant-data",
      name: "tag-test2",
    });
  });

  it("Fails to create tag", async () => {
    wrapper.vm.showDialog = true;

    await flushPromises();

    mockTags.onPost("http://localhost:3000/api/namespaces/fake-tenant-data/tags").reply(409);

    await wrapper.findComponent('[data-test="tag-field"]').setValue("");

    await wrapper.findComponent('[data-test="create-btn"]').trigger("click");

    await flushPromises();

    expect(mockSnackbar.showError).toHaveBeenCalledWith("Failed to create tag.");
  });
});
