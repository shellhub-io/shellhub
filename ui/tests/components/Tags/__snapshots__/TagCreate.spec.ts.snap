// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tag Form Create > Renders the component 1`] = `""`;
